### PR TITLE
Update to web3 1.0

### PR DESCRIPTION
**Pull request by cgewecke**
Misc. method translation, conversion to promise from callback, and BN methods. 

Also added some file cleanup logic to the tests.

### CODE REVIEW

File: lib/test.js [hunk 1]
```diff
@@ -149,15 +149,6 @@ var Test = {
...
   },
 
-  getAccounts: function(web3, config) {
-    return new Promise(function(accept, reject) {
-      web3.eth.getAccounts(function(err, accs) {
-        if (err) return reject(err);
-        accept(accs);
-      });
-    });
-  },
```
Comment (gnidan) [inline lib/test.js]: Good riddance

File: test/profiler.js [hunk 2]
```diff
@@ -21,6 +23,14 @@ describe('profiler', function() {
     });
   });
 
+  after("Cleanup tmp files", function(done){
+    glob('tmp-*', (err, files) => {
+      if(err) done(err);
+      files.forEach(file => fs.removeSync(file));
+      done();
+    })
+  })
```
Comment (gnidan) [inline test/profiler.js]: There's gotta be a better way to make sure those get cleaned up. Well, whatev.

Comment (cgewecke) [reply test/profiler.js]: Add it the npm script? ` && rm -rf tmp-*`

Comment (gnidan) [reply test/profiler.js]: It's probably not worth the change right now, but I like that idea

**Pull request merged**